src/main/groovy/com/twcable/grabbit/spring/batch/repository/JcrGrabbitJobInstanceDao.groovy

```diff
@@ -257,10 +257,21 @@ class JcrGrabbitJobInstanceDao extends AbstractJcrDao implements GrabbitJobInsta
         if (!resolver) throw new IllegalArgumentException("resolver == null")
 
         final rootResource = resolver.getResource(JOB_INSTANCE_ROOT)
-        final nextId = (rootResource.children.asList().size() + 1) as Long
+
+        final lastInstance = rootResource?.children?.asList()?.max { Resource resource ->
+            final properties = resource.adaptTo(ValueMap)
+            properties[INSTANCE_ID] as Long
+        }
+        final lastInstanceProperties = lastInstance?.adaptTo(ValueMap)
+        Long nextId = lastInstanceProperties?.get(INSTANCE_ID) as Long
+        if (!nextId) {
+            nextId = 1
+        } else {
+            nextId += 1
+        }
```

Comment:
As @jdigger has pointed out, the previous code contained a concurrency issue which this update does not address.  A simple way to address the concurrency issue may be to modify the createJobInstance method (which calls this method to get a new jobInstanceId) to catch the exception that will occur when we try to create two instances with the same ID.  Then just call this function to get a new Id and retry the create.